packages/vkui/src/hooks/useResizeObserver.ts

```diff
@@ -0,0 +1,29 @@
+import * as React from 'react';
+import { CustomResizeObserver } from '../lib/floating/customResizeObserver';
+import { useIsomorphicLayoutEffect } from '../lib/useIsomorphicLayoutEffect';
+import { useStableCallback } from './useStableCallback';
+
+/**
+ * Хук вызывает переданный коллбэк при изменении размеров элемента.
+ */
+export function useResizeObserver(
+  ref: React.MutableRefObject<HTMLElement | null>,
+  callback: (element: HTMLElement) => void,
+) {
+  const stableCallback = useStableCallback(callback);
+
+  useIsomorphicLayoutEffect(
+    function addResizeObserverHandler() {
+      if (!ref.current) {
```

Comment:
чтобы убрать из покрытия

```suggestion
      /* istanbul ignore if: невозможный кейс (в SSR вызова этой функции не будет) */
      if (!ref.current) {
```

пример, https://github.com/VKCOM/VKUI/blob/v6.1.1/packages/vkui/src/lib/animation/useReducedMotion.ts#L22